test(coverage/typescript): Add reviewed tsc diagnostics error codes

leaysgur opened the pull request:
Finally fixes #11582 

This PR adds a script that generates a list of TSC error codes.

The codes are collected based on the TypeScript version managed by `just submodules`.
And the list is then committed to Git.

When the submodule version is updated, the script is also called and the list is regenerated.

If TSC introduces new diagnostic codes, the diff will show the added code, which we need to review.
Either leave it for the parser to report or ignore it if we cannot support it.

leaysgur commented:
* **#15542** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15542?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15542?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15542).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen reviewed (commented):
I'm not comfortable maintaining another submodule 😢 please let me study what this PR is actually doing first.

leaysgur committed:
test(coverage/typescript): Add reviewed tsc diagnostics error codes (#15542)

Finally fixes #11582

This PR adds a script that generates a list of TSC error codes.

The codes are collected based on the TypeScript version managed by `just submodules`.
And the list is then committed to Git.

When the submodule version is updated, the script is also called and the list is regenerated.

If TSC introduces new diagnostic codes, the diff will show the added code, which we need to review.
Either leave it for the parser to report or ignore it if we cannot support it.

Boshen reviewed (commented):
This should be part of the conformance runner instead of separate js task, which I assume is slow.

I'll get claude to work on this at some point.
